### PR TITLE
Fixes processing multiple k8s resource artifacts

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes: https://github.com/projectatomic/atomicapp/issues/212

 - Nulecule multiple resource artifacts defined per kind of resource can now be
 processed without overriding earlier artifacts
 - Modifies the methods names as per <https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables>
 - Alters order of imports as per <https://www.python.org/dev/peps/pep-0008/#imports>
 - Adds docstring to methods

### CODE REVIEW

File: atomicapp/providers/kubernetes.py [hunk 1]
```diff
@@ -128,9 +147,14 @@ def _resourceIdentity(self, path):
...
-        rname = self._resourceIdentity(path)
-        cmd = [self.kubectl, "resize", "rc", rname, "--replicas=0", "--namespace=%s" %
+    def _scale_replicas(self, path, replicas=0):
+        """Scales replicationController to specified replicas size
+
+        :arg path: Path to replicationController manifest
+        :arg replicas: Replica size to scale to.
+        """
+        rname = self._resource_identity(path)
+        cmd = [self.kubectl, "scale", "rc", rname, "--replicas=0", "--namespace=%s" %
```
Comment (navidshaikh) [inline atomicapp/providers/kubernetes.py]: With latest `v1` APIs, `scale` is introduced and recommended to be used.

**Comment by vpavlin** [comment]
From what I was told earlier the order of manifest kinds matter (i.e. first you deploy `service`, then `rc` and/or `pod`). Brief look at your code gives me an impression that this order is not ensured.

1. Is the order requirement still the case for kubes?
2. Am I right with the assumption that the order went away with your path?

**Comment by navidshaikh** [comment]
@vpavlin : 
I think the order in which the resources should be deployed must be ensured by the Nulecule author and IMO we should not alter the sequence of the k8s resource listing and their deployment.

File: atomicapp/providers/kubernetes.py [hunk 2]
```diff
@@ -105,18 +112,30 @@ def _callK8s(self, path):
                 printErrorStatus("cmd failed: " + " ".join(cmd))
                 raise
 
-    def prepareOrder(self):
+    def process_k8s_artifacts(self):
+        """Processes Kubernetes manifests files and checks if manifest under
```
Comment (goern) [inline atomicapp/providers/kubernetes.py]: this is 'manifest after processing' ?

File: atomicapp/providers/kubernetes.py [hunk 3]
```diff
@@ -128,9 +147,14 @@ def _resourceIdentity(self, path):
         else:
             raise ProviderFailedException("Malformed kube file: %s" % path)
 
-    def _resetReplicas(self, path):
-        rname = self._resourceIdentity(path)
-        cmd = [self.kubectl, "resize", "rc", rname, "--replicas=0", "--namespace=%s" %
+    def _scale_replicas(self, path, replicas=0):
```
Comment (goern) [inline atomicapp/providers/kubernetes.py]: this is not _scale_replicas() it is _set_replica_count_to_zero()

File: atomicapp/providers/kubernetes.py (hunk 1 above)
Comment (goern) [inline atomicapp/providers/kubernetes.py]: at least here is a hardcoded 0

**Comment by goern** [comment]
I think we dont need any sequence in deploying to kubernetes or openshift.

**Comment by vpavlin** [comment]
As I said, originally there was a requirement for deploying service before pods/rcs

File: atomicapp/providers/kubernetes.py (hunk 3 above)
Comment (navidshaikh) [inline atomicapp/providers/kubernetes.py]: Correct!
Fixed in next patch.

File: atomicapp/providers/kubernetes.py (hunk 2 above)
Comment (navidshaikh) [inline atomicapp/providers/kubernetes.py]: Sorry, I did not get this!

**Comment by aweiteka** [comment]
LGTM

**Comment by navidshaikh** [comment]
Added new atomicapp example of gitlab app for the fix of this issue. This atomicapp app particularly references multiple Kubernetes artifacts for 3 different tiers and the tests perform `install`, `run` and `stop` operations on them.

**Comment by vpavlin** [comment]
Thanks @navidshaikh! LGTM

**Pull request merged**